### PR TITLE
Delta client secret #40

### PR DESCRIPTION
**Pull request by numbat**

### CODE REVIEW

File: src/main/python/covata/delta/client.py [hunk 1]
```diff
@@ -89,16 +89,66 @@ def get_identity(self, identity_id, identity_to_retrieve=None):
...
+
+        :param str identity_id: the authenticating identity id
+        :param bytes content: the secret contents
+        :return: the secret
+        :rtype: :class:`~.Secret`
+        """
+        secret_key = crypto.generate_secret_key()
+        iv = crypto.generate_initialisation_vector()
+
+        crypto.encrypt(content, secret_key, iv)
```
Comment (myuwono) [inline src/main/python/covata/delta/client.py]: encrypt should've returned the cihpertext and the tag..?

File: src/main/python/covata/delta/client.py [hunk 2]
```diff
@@ -150,3 +201,96 @@ def external_id(self):
...
+        Creates a new secret with the given parameters.
+
+        :param parent: the Delta client that constructed this instance
+        :param str id: the id of the secret
+        :param str created: the created date
+        :param str rsa_key_owner: the identity id of the RSA key owner
+        :param str created_by: the identity id of the secret creator
+        :param encryption_details: the encryption details of the secret
+        :type parent: :class:`~.Client`
+        :type encryption_details: :class:`~.'EncryptionDetails`
```
Comment (myuwono) [inline src/main/python/covata/delta/client.py]: typo `~.'Encr...`

File: src/main/python/covata/delta/client.py [hunk 3]
```diff
@@ -89,16 +89,66 @@ def get_identity(self, identity_id, identity_to_retrieve=None):
...
+
+        :param str identity_id: the authenticating identity id
+        :param str secret_id: the id of the secret to retrieve
+        :return: the secret
+        :rtype: :class:`~.Secret`
+        """
+        response = self.api_client.get_secret(identity_id, secret_id)
+
+        return Secret(self,
+                      response.get("id"),
```
Comment (myuwono) [inline src/main/python/covata/delta/client.py]: we can use square brackets to access `dict` contents

File: src/test/python/test_client.py [hunk 4]
```diff
@@ -31,6 +35,21 @@ def api_client(key_store):
     return ApiClient(key_store)
 
 
+@pytest.fixture(scope="function")
+def mock_crypto(mocker):
```
Comment (myuwono) [inline src/test/python/test_client.py]: mock_crypto is unused?

Comment (numbat) [inline src/test/python/test_client.py]: Good catch - now used, was defaulting back to actual crypto calls.

**Pull request merged**